### PR TITLE
Acts-Svtx Object Merging

### PR DESCRIPTION
**Pull request by osbornjd**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [X] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people) (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

This PR introduces a major change to the track and cluster framework. The PR adds in the necessary information to the `TrkrCluster` and `SvtxTrack` objects to build Acts compatible objects on the fly when requested. The changes will then allow for Acts::SourceLinks and Acts::TrackParameters to be created directly from `TrkrCluster` or `SvtxTrack` objects, which will eliminate the need for `PHActsSourceLinks` and `PHActsTracks`. 

A future PR will modify the `trackreco` package to allow this to be the case - however, for the time being, the track reconstruction chain is left unmodified in this sense. This PR only creates the framework necessary for that to happen.

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

## Links to other PRs in macros and calibration repositories (if applicable)

Related [PR](https://github.com/sPHENIX-Collaboration/macros/pull/385) in macros which must be merged in conjunction with this to avoid any breaking changes.

### CODE REVIEW

**Comment by osbornjd** [comment]
BTW, there are other problems too - now that Acts is infiltrating the basic sPHENIX TrkrCluster and SvtxTrack classes I think that anything less than gcc8.3 can't compile by definition. I'm not sure how we get around that either

File: offline/packages/micromegas/MicromegasClusterizer.cc [hunk 1]
```diff
@@ -303,6 +307,61 @@ int MicromegasClusterizer::process_event(PHCompositeNode *topNode)
         break;
       }
 
+      /// Add Acts and local information
+      const unsigned int layer = TrkrDefs::getLayer(cluster_key);
+      unsigned int tile = 0;
+      MicromegasDefs::SegmentationType segtype;
```
Comment (hupereir) [inline offline/packages/micromegas/MicromegasClusterizer.cc:313]: Hi Joe, 
I don't think you need to retrieve again the segmentation type here (and in particular not make it hard coded to layer number). There is already a segmentation type defined at line 147, which is retrieved from the layer geometry, and should still be available, and valid here. Can you double check ?

File: offline/packages/micromegas/MicromegasClusterizer.cc [hunk 2]
```diff
@@ -303,6 +307,61 @@ int MicromegasClusterizer::process_event(PHCompositeNode *topNode)
...
+      if(layer == 55)
+	segtype  =  MicromegasDefs::SegmentationType::SEGMENTATION_PHI;
+      else
+	segtype = MicromegasDefs::SegmentationType::SEGMENTATION_Z;
+      
+      Acts::Vector3D globalPos(cluster->getX(), cluster->getY(), cluster->getZ());
+          
+      /// Get the surface key to find the surface from the map
+      TrkrDefs::hitsetkey mmHitSetKey = MicromegasDefs::genHitSetKey(layer, segtype, tile);
```
Comment (hupereir) [inline offline/packages/micromegas/MicromegasClusterizer.cc:322]: Same remark, there is already a tileid and an hitsetkey from lines 135 and 137. Can you reuse those ? 
In particular as far as I can see your "tile" here is always zero, so the hitsetkey (and surface) will actually not match. Or do I miss something ?

**Comment by osbornjd** [comment]
This PR has been developed on top of and at this point is obsolete. I'm just going to close it and open a new one with additional features

**Pull request closed without merging**